Doxygen warnings

There are some doxygen warnings:

```
Generating docs for file dec_png.c...
/home/ed/NCEPLIBS-g2/src/dec_png.c:43: warning: parameters of member user_read_data are not (all) documented
Generating docs for file drstemplates.f...
Generating docs for file enc_jpeg2000.c...
Generating docs for file enc_png.c...
Generating docs for file g2_gbytesc.f...
Generating docs for file g2grids.f...
Generating docs for file gb_info.f...
Generating docs for file gdt2gds.f...
Generating docs for file getdim.f...
Generating docs for file getfield.f...
Generating docs for file getg2i.f...
Generating docs for file getg2ir.f...
Generating docs for file getgb2.f...
Generating docs for file getgb2l.f...
Generating docs for file getgb2p.f...
Generating docs for file getgb2r.f...
Generating docs for file getgb2rp.f...
Generating docs for file getgb2s.f...
Generating docs for file getidx.f...
Generating docs for file getlocal.f...
Generating docs for file getpoly.f...
Generating docs for file gettemplates.f...
Generating docs for file gf_free.f...
/home/ed/NCEPLIBS-g2/src/gf_free.f:138: warning: argument 'gfld' from the argument list of gf_free has multiple @param documentation sections
Generating docs for file gf_getfld.f...
Generating docs for file gf_unpack1.f...
Generating docs for file gf_unpack2.f...
Generating docs for file gf_unpack3.f...
Generating docs for file gf_unpack4.f...
Generating docs for file gf_unpack5.f...
Generating docs for file gf_unpack6.f...
Generating docs for file gf_unpack7.f...
/home/ed/NCEPLIBS-g2/src/gf_unpack7.f:42: warning: argument 'ndpts' from the argument list of gf_unpack7 has multiple @param documentation sections
Generating docs for file gribcreate.f...
Generating docs for file gribend.f...
Generating docs for file gribinfo.f...
Generating docs for file gribmod.f...
/home/ed/NCEPLIBS-g2/src/gribmod.f:153: warning: Member g2_version (variable) of module grib_mod is not documented.
Generating docs for file gridtemplates.f...
Generating docs for file intmath.f...
Generating docs for file ixgb2.f...
Generating docs for file jpcpack.f...
Generating docs for file jpcunpack.f...
Generating docs for file misspack.f...
Generating docs for file mkieee.f...
Generating docs for file mova2i.c...
Generating docs for file pack_gp.f...
Generating docs for file params.f...
Generating docs for file params_ecmwf.f...
Generating docs for file pdstemplates.f...
Generating docs for file pngpack.f...
Generating docs for file pngunpack.f...
Generating docs for file putgb2.f...
Generating docs for file rdieee.f...
Generating docs for file realloc.f...
Generating docs for file reduce.f...
Generating docs for file simpack.f...
Generating docs for file simunpack.f...
Generating docs for file skgb.f...
Generating docs for file specpack.f...
Generating docs for file specunpack.f...
Generating page documentation...
Generating group documentation...
Generating class documentation...
Generating namespace index...
Generating docs for namespace drstemplates
/home/ed/NCEPLIBS-g2/src/drstemplates.f:121: warning: End of list marker found without any preceding list items
Generating docs for namespace g2grids
Generating docs for namespace grib_mod
/home/ed/NCEPLIBS-g2/src/gribmod.f:153: warning: Member g2_version (variable) of module grib_mod is not documented.

```

After changing back to doxygen 1.8.17 as is used on the CI system, I get these instead:

```

/home/runner/work/NCEPLIBS-g2/NCEPLIBS-g2/g2/src/gf_free.f:138: warning: argument 'gfld' from the argument list of gf_free has multiple @param documentation sections
/home/runner/work/NCEPLIBS-g2/NCEPLIBS-g2/g2/src/gf_unpack7.f:42: warning: argument 'ndpts' from the argument list of gf_unpack7 has multiple @param documentation sections
/home/runner/work/NCEPLIBS-g2/NCEPLIBS-g2/g2/src/gribmod.f:153: warning: Member g2_version (variable) of module grib_mod is not documented.
/home/runner/work/NCEPLIBS-g2/NCEPLIBS-g2/g2/src/drstemplates.f:121: warning: End of list marker found without any preceding list items
/home/runner/work/NCEPLIBS-g2/NCEPLIBS-g2/g2/src/gribmod.f:153: warning: Member g2_version (variable) of module grib_mod is not documented.
```

--- a/docs/Doxyfile.in
+++ b/docs/Doxyfile.in
@@ -791,7 +791,7 @@ WARN_NO_PARAMDOC       = YES
 # a warning is encountered.
 # The default value is: NO.
 
-WARN_AS_ERROR          = NO
+WARN_AS_ERROR          = YES
 
 # The WARN_FORMAT tag determines the format of the warning messages that doxygen
 # can produce. The string should contain the $file, $line, and $text tags, which
@@ -1005,13 +1005,13 @@ STRIP_CODE_COMMENTS    = YES
 # entity all documented functions referencing it will be listed.
 # The default value is: NO.
 
-REFERENCED_BY_RELATION = NO
+REFERENCED_BY_RELATION = YES
 
 # If the REFERENCES_RELATION tag is set to YES then for each documented function
 # all documented entities called/used by that function will be listed.
 # The default value is: NO.
 
-REFERENCES_RELATION    = NO
+REFERENCES_RELATION    = YES
 
 # If the REFERENCES_LINK_SOURCE tag is set to YES and SOURCE_BROWSER tag is set
 # to YES then the hyperlinks from functions in REFERENCES_RELATION and

--- a/src/dec_png.c
+++ b/src/dec_png.c
@@ -40,13 +40,15 @@ typedef struct png_stream png_stream; /**< location to write PNG stream */
 
 void user_read_data(png_structp , png_bytep , png_uint_32 );
 
-void user_read_data(png_structp png_ptr,png_bytep data, png_uint_32 length)
-
 /**
  * Custom read function used so that libpng will read a PNG stream
  * from memory instead of a file on disk.
+ *
+ * @param png_ptr pointer to png.
+ * @param data the data.
+ * @param length length of data.
 */
-
+void user_read_data(png_structp png_ptr,png_bytep data, png_uint_32 length)
 {
      char *ptr;
      g2int offset;

--- a/src/drstemplates.f
+++ b/src/drstemplates.f
@@ -112,11 +112,9 @@
 
       contains
 
-!>    @brief  This function returns the index of specified Data
+!>    This function returns the index of specified Data
 !>    Representation Template 5.NN (NN=number) in array templates.                                    .
-!>    @author Stephen Gilbert            @date 2001-06-28
 !>     
-
 !>    @param[in] number NN, indicating the number of the Data Representation 
 !>    Template 5.NN that is being requested.
 !>
@@ -139,11 +137,9 @@
 
          end function
 
-!>    @brief This subroutine returns DRS template information for a                                   .
+!>    This subroutine returns DRS template information for a                                   .
 !>    specified Data Representation Template 5.NN.
-!>    @author Stephen Gilbert            @date 2000-05-11
-!>     
-
+!>
 !>    The number of entries in the template is returned along with a map
 !>    of the number of octets occupied by each entry. Also, a flag is
 !>    returned to indicate whether the template would need to be extended.
@@ -184,11 +180,9 @@
 
          end subroutine
 
-!>    @brief This subroutine generates the remaining octet map for a
+!>    This subroutine generates the remaining octet map for a
 !>    given Data Representation Template, if required.
-!>    @author  Stephen Gilbert            @date 2000-05-11
 !>     
-
 !>    Some Templates can vary depending on data values given in an earlier part 
 !>    of the Template, and it is necessary to know some of the earlier entry
 !>    values to generate the full octet map of the Template.

--- a/src/drstemplates.f
+++ b/src/drstemplates.f
@@ -113,7 +113,7 @@
       contains
 
 !>    This function returns the index of specified Data
-!>    Representation Template 5.NN (NN=number) in array templates.                                    .
+!>    Representation Template 5.NN (NN=number) in array templates.
 !>     
 !>    @param[in] number NN, indicating the number of the Data Representation 
 !>    Template 5.NN that is being requested.
@@ -137,7 +137,7 @@
 
          end function
 
-!>    This subroutine returns DRS template information for a                                   .
+!>    This subroutine returns DRS template information for a
 !>    specified Data Representation Template 5.NN.
 !>
 !>    The number of entries in the template is returned along with a map

--- a/src/gf_free.f
+++ b/src/gf_free.f
@@ -10,8 +10,7 @@
 !>    - 2000-05-26 Stephen Gilbert Modified from getg1s to work with grib2
 !>    - 2012-12-11 Boi Vuong initialize an undefine pointers
 !>    - 2015-10-29 Boi Vuong Deallocate pointers in derived type gribfield
-!>    @param[in] gfld derived type gribfield (defined in module grib_mod)
-!>    @param[out] gfld derived type gribfield (defined in module grib_mod)
+!>    @param gfld derived type gribfield (defined in module grib_mod)
 !>    (NOTE: See Remarks Section)
 !>    - gfld\%version GRIB edition number (currently 2)
 !>    - gfld\%discipline Message Discipline (see Code Table 0.0)

--- a/src/gf_unpack7.f
+++ b/src/gf_unpack7.f
@@ -20,7 +20,6 @@
 !>    for the specified Grid Definition. Template (N=igdsnum). Each element of this
 !>    integer array contains an entry (in the order specified) of Grid Definition
 !>    Template 3.N (Only required to unpack DRT 5.51).
-!>    @param[in] ndpts Number of data points unpacked and returned.
 !>    @param[in] idrsnum Data Representation Template Number (Code Table 5.0).
 !>    @param[in] idrstmpl Pointer to an integer array containing the data values for
 !>    the specified Data Representation Template (N=idrsnum). Each element of this 

--- a/src/gribmod.f
+++ b/src/gribmod.f
@@ -150,7 +150,7 @@
 
       module grib_mod
 
-      character(len=12) :: G2_VERSION="g2lib-3.1.0"
+      character(len=12) :: G2_VERSION="g2lib-3.5.0" !< Library version.
 
       type gribfield
           integer :: version,discipline

